h2o-r/h2o-package/R/frame.R

```diff
@@ -3782,6 +3782,67 @@ h2o.arrange <- function(x, ...) {
   .newExpr("sort", x, by-1L, as.numeric(ascend))
 }
 
+generateColInd <-function(data, by) {
+  ### handle the columns
+  # we accept: c('col1', 'col2'), 1:2, c(1,2) as column names.
+  if(base::is.character(by)) {
+    group.cols <- match(by, colnames(data))
+    if (any(is.na(group.cols)))
+    stop('No column named ', by, ' in ', substitute(data), '.')
+  } else if(is.integer(by)) {
+    group.cols <- by
+  } else if(is.numeric(by)) {   # this will happen eg c(1,2,3)
+    group.cols <- as.integer(by)
+  }
+
+  if(group.cols <= 0L || group.cols > ncol(data)) {
+   stop('Column ', group.cols, ' out of range for frame columns ', ncol(data), '.')
+  }
+
+  if (anyDuplicated(by)) stop("Some duplicate column names have been provided")
+
+  return(group.cols)
+}
+
```

Comment:
The first comment line should be a short summary, like "Ranks rows within a GroupBy object" (or whatever makes sense). Then this becomes the function description in the docs. Right now, this shows up as a very large paragraph for the function summary, and the "Description" is only the "To sort column c1..." paragraph. (Which could be a good paragraph for the "Details" section.)

Reply:
@wendycwong See my comment above, same as @angela0xdata.